### PR TITLE
Ps 117 post comment

### PR DESCRIPTION
**Pull request by Tibor22**
Includes  #116 , #115 issues. Also, the optional property on updatedAt was removed.

### CODE REVIEW

**Comment by dearshrewdwit** [comment]
Nice. There were some outstanding questions on the previous PR for this: 
> https://github.com/boolean-uk/team-dev-server/pull/117

Have they been answered/resolved?

**Comment by Hoolio333** [comment]
> Nice. There were some outstanding questions on the previous PR for this:
> 
> > #117
> 
> Have they been answered/resolved?

Yes all resolved and working.

File: src/controllers/post.js [hunk 1]
```diff
@@ -10,11 +11,40 @@ export const create = async (req, res) => {
...
+    commentToCreate.profileId = req.user.id
+    commentToCreate.postId = postId
+    const comment = await commentToCreate.save()
+    return sendDataResponse(res, 201, comment)
+  } catch (err) {
+    return sendDataResponse(res, 400, { err: err.message })
+  }
+}
+export const findAllComments = async (req, res) => {
+  try {
```
Comment (dearshrewdwit) [inline src/controllers/post.js:39]: What's the reasoning for the try/catch block here? What might throw an error?

File: src/controllers/post.js [hunk 2]
```diff
@@ -10,11 +11,40 @@ export const create = async (req, res) => {
...
   } catch (err) {
     return sendDataResponse(res, 400, { err: err.message })
   }
 }
+
+export const createComment = async (req, res) => {
+  const postId = +req.query.postId
+  const { content } = req.body
+  if (!content) {
+    return sendDataResponse(res, 400, { content: 'Must provide content' })
```
[the remark this answers is not in the document]
Comment (dearshrewdwit) [reply src/controllers/post.js:25]: It doesn't matter that much - the function in `src/utils/responses.js` simply passes it back to the client

**Comment by dearshrewdwit** [comment]
> > Nice. There were some outstanding questions on the previous PR for this:
> > > #117
> > 
> > 
> > Have they been answered/resolved?
> 
> Yes all resolved and working.

The original questions are still unanswered - or can you point me to where they've been discussed?

File: src/controllers/post.js (hunk 1 above)
Comment (Tibor22) [reply src/controllers/post.js:39]: Originally was not working but we modified the try block to throw a new error if no comments are found.

Comment (Hoolio333) [reply src/controllers/post.js:39]: Catch error insomnia

**Comment by dearshrewdwit** [comment]
> > > > Nice. There were some outstanding questions on the previous PR for this:
> > > > > #117
> > > > 
> > > > 
> > > > Have they been answered/resolved?
> > > 
> > > 
> > > Yes all resolved and working.
> > 
> > 
> > The original questions are still unanswered - or can you point me to where they've been discussed?
> 
> Please can you clarify these questions as we can't find them?

here: https://github.com/boolean-uk/team-dev-server/pull/117

**Comment by shahzadnaeem** [comment]
This PR needs to be updated as changes to the DB migration #116 are required to ensure the addition of the 'updatedAt' column does not fail for existing rows where this column is added as the column currently has a 'NOT NULL' constraint.

The fix is to split the migration into three consecutive steps:

1. Add the new column without the 'NOT NULL' constraint
2. Update all rows in all affected tables to add a value for the new column
3. Finally, add the 'NOT NULL' constraint - this should not fail (although there is still a possibility it may ... not sure is multiple migrations can be set to be run in a single transaction with Prisma)

**Review by dearshrewdwit — APPROVED** [review]

**Review by d-username — APPROVED** [review]

**Pull request merged**